src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs

```diff
@@ -314,7 +314,15 @@ internal virtual IEnumerable<KeyValuePair<string, DType>> GetArgumentSuggestions
         /// <returns>
         /// The suggestion kind for the hypothetical suggestion.
         /// </returns>
-        internal virtual SuggestionKind GetFunctionSuggestionKind(TexlFunction function, int argumentIndex) => SuggestionKind.Global;
+        internal virtual SuggestionKind GetFunctionSuggestionKind(TexlFunction function, int argumentIndex)
+        {
+            if (function.HasTypeArgs && function.ArgIsType(argumentIndex))
```

Comment:
>unction.HasTypeArgs && [](http://example.com/codeflow?start=17&length=22)

Is this needed? If the second term of the `&&` is true this has to be true as well.

Reply:
fixed. since we have that contract , it makes sense not to repeat the test